src/main/java/frc/robot/subsystems/shooter/LinearRegression.java

```diff
@@ -0,0 +1,100 @@
+package frc.robot.subsystems.shooter;
+
+import frc.robot.Constants;
+
+import java.io.*;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * The linear regression contains a list of velocity based on the distance, and should give an approximation of the velocity to apply.
+ * In order to the linear regression object to approximate better, you need to pass more values.
+ * The object calculate calculate a straight line between the closest point, and retrieve the velocity (in the straight line between the records).
+ */
+public class LinearRegression {
```

Comment:
Actually it isn't a linear regression. In linear regression there is one function for all the point, here you create a function between each 2 points which is more like moving average

Reply:
Moving average is used when you want to smooth out some values, here you have two types of values. From the Wikipedia article on ["Simple linear regression"](https://en.wikipedia.org/wiki/Simple_linear_regression):
> predicts the dependent variable values as a function of the independent variable.

Which is what you do here. Indeed, you don't have one function, but you've just simplified the regression and built a series of linear functions.

Reply:
If we create a series of linear function by a defined distance on the x axis it's mean we made a moving average

Reply:
Maybe, technically, yes. But in concept, the idea is to predict the velocity from the distance, and that's what linear regressions do.

Reply:
Also moving average is ment for predictions in stock market to spot trends in the graph

Reply:
Maybe we can agree on moving regression🤔?

Reply:
> Also moving average is ment for predictions in stock market to spot trends in the graph

I'm no expert when it comes to stock markets, but I guess they are used so _if_ you have a trend (for example a short surge in a stock price) it will smooth out, so I don't see how it's related to predictions (but I'd love to hear from you otherwise if I'm wrong).

> Maybe we can agree on moving regression🤔?

lol, that could work. But it's your call, you're (literally) the boss.